WordPress/Classes/Services/SignupService.swift

```diff
@@ -1,84 +1,74 @@
-                                            defer {
-                                                WPAppAnalytics.track(.createdAccount, withProperties: ["source": "google"])
-                                            }
-                                            success(createdAccount)
-                                        } else {
-                                            defer {
-                                                WPAppAnalytics.track(.signupSocialToLogin)
-                                            }
-                                            // we need to sync the blogs for existing accounts to be able to display the Login Epilogue
-                                            BlogSyncFacade().syncBlogs(for: account, success: {
-                                                success(createdAccount)
-                                            }, failure: { (_) in
-                                                // the blog sync failed but the user is already logged in
-                                                success(createdAccount)
-                                            })
-                                        }
-                                    },
-                                    failure: failure)
+        remote.createWPComAccount(withGoogle: googleToken,
+                                  andLocale: locale,
+                                  andClientID: configuration.wpcomClientId,
+                                  andClientSecret: configuration.wpcomSecret,
+                                  success: { response in
+
+            guard let username = response?[ResponseKeys.username] as? String,
+                let bearer_token = response?[ResponseKeys.bearerToken] as? String else {
+                    failure(ServiceError.unknown)
+                    return
+            }
+
+            let createdAccount = (response?[ResponseKeys.createdAccount] as? Int ?? 0) == 1
+            success(createdAccount, username, bearer_token)
+        }, failure: { error in
+            failure(error ?? ServiceError.unknown)
+        })
     }
+}
 
 
+// MARK: - Private
+//
+private extension SignupService {
 
-    // MARK: Private Instance Methods
-
-    @objc func anonymousApi() -> WordPressComRestApi {
-        return WordPressComRestApi(userAgent: WPUserAgent.wordPress())
+    private var anonymousAPI: WordPressComRestApi {
+        return WordPressComRestApi(oAuthToken: nil, userAgent: configuration.userAgent)
     }
 
+    private var configuration: WordPressAuthenticatorConfiguration {
+        return WordPressAuthenticator.shared.configuration
+    }
 
-    /// A convenience struct for response keys
     private struct ResponseKeys {
         static let bearerToken = "bearer_token"
         static let username = "username"
         static let createdAccount = "created_account"
     }
 }
+
+
+// MARK: - Errors
+//
+extension SignupService {
+
+    enum ServiceError: Error {
```

Comment:
This exists three times in the code. Any reason not to move it to be defined just once?